### PR TITLE
[Android][Task][56449] Init base component

### PR DESCRIPTION
**Pull request by hungvv-1301**
## Related Tickets
- [#Task #56449: Setup Retrofit, handle call API](https://edu-redmine.sun-asterisk.vn/issues/56449)
- [#Task #56450: Setup Rx](https://edu-redmine.sun-asterisk.vn/issues/56450)
- [#Task #56452: Setup DI , Bottom naigation](https://edu-redmine.sun-asterisk.vn/issues/56452)
## WHAT
- Set up project
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -7,4 +7,11 @@
     <color name="teal_700">#FF018786</color>
     <color name="black">#FF000000</color>
     <color name="white">#FFFFFFFF</color>
+
+    <color name="main_500">#7FD679</color>
+    <color name="main">#71AA6D</color>
+    <color name="main_700">#499144</color>
+    <color name="main_900">#317C2C</color>
+    <color name="red">#F44336</color>
+    <color name="gray">#7B8E8E</color>
```
Comment (hungld-1302) [inline app/src/main/res/values/colors.xml]: thêm color_ ở trước

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <dimen name="_5dp">5dp</dimen>
     <dimen name="_6dp">6dp</dimen>
     <dimen name="_80dp">80dp</dimen>
+    <dimen name="_100dp">100dp</dimen>
 
     <dimen name="margin_4dp">4dp</dimen>
     <dimen name="margin_6dp">6dp</dimen>
```
Comment (hungld-1302) [inline app/src/main/res/values/dimens.xml:12]: sort?

File: app/src/main/java/com/atom/android/lebo/data/di/NetworkModule.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+private fun provideRetrofit(client: OkHttpClient): Retrofit {
+    return Retrofit.Builder()
+        .baseUrl(BuildConfig.API_HOST)
+        .addConverterFactory(GsonConverterFactory.create())
+        .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+        .client(client)
+        .build()
+}
```
Comment (ghost) [inline app/src/main/java/com/atom/android/lebo/data/di/NetworkModule.kt:51]: thêm HttpLoggingInterceptor vào debug cho tiện

File: app/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -7,4 +7,11 @@
     <color name="teal_700">#FF018786</color>
     <color name="black">#FF000000</color>
     <color name="white">#FFFFFFFF</color>
+
+    <color name="main_500">#7FD679</color>
+    <color name="main">#71AA6D</color>
+    <color name="main_700">#499144</color>
+    <color name="main_900">#317C2C</color>
```
Comment (ndt0409) [inline app/src/main/res/values/colors.xml]: màu ?

File: app/src/main/java/com/atom/android/lebo/utils/extensions/RxExtension.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+
+fun Completable.withIOToMainThread(): Completable =
+    subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+
+fun <T> Observable<T>.prepareValidate() =
+    debounce(Constant.DELAY.INPUT_TEXT, TimeUnit.MILLISECONDS).withIOToMainThread()
+
+fun <T> Single<T>.handleLoading(show: () -> Unit, hide: () -> Unit) =
+    doOnSubscribe { show() }.doOnSuccess { hide() }.doOnError { hide() }
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/utils/extensions/RxExtension.kt]: đặt tên các hàm common thôi e , sao lại có prepareValidate ở đây , nếu e muốn add debounce vào observable thì đặt tên kiểu , ignoreFastAction ,...

**Pull request merged**